core/trino-spi/src/main/java/io/trino/spi/connector/ColumnMetadata.java

```diff
@@ -135,6 +135,11 @@ public boolean isHidden()
         return properties;
     }
 
+    public ColumnBasicMetadata toBasicMetadata()
+    {
+        return new ColumnBasicMetadata(name, type, nullable, hidden);
```

Comment:
If the intention is that `ColumnBasicMetadata` is a subset of information, maybe `ColumnBasicMetadata` should be accepted in the `ColumnMetadata` ctor and be stored as a field? This could allow connectors to maintain mapping once.

However, I am not sure whether we can want to lock ourselves in one being subset of the other -- logically it make sense, but there can be a situation, where this isn't helpful. For this however, we would want this method (`toBasicMetadata()`) not to exist at all. (or exist as deprecated for the transition period, if needed)

thoughts?

Reply:
> If the intention is that ColumnBasicMetadata is a subset of information, maybe ColumnBasicMetadata should be accepted in the ColumnMetadata ctor and be stored as a field?

Sounds good to me. That was my intention. I wanted to leave it optional for connectors to provide basic metadata. For example in case of Hive it does not really matter if we provide basic or non-basic as it translates to same number of HMS calls.

Reply:
> be stored as a field

That was my plan too. Also @martint mentioned that is the way he would prefer. However it is not that simple as there is not only ConnectorTableSchema, but also ColumnSchema. So I would need to somehow split `ColumnMetadata` in `ConnectorTableMetadata`.